freeze Werkzeug version at 2.3.7

matt-yml opened the pull request:
## Summary
Fix compatibility issue between Flask 2.1.1 and Werkzeug 3.x.x

## Description
This pull request addresses the runtime error encountered with Flask 2.1.1 due to incompatibility with certain Werkzeug versions. It resolves the following error:
`ImportError: cannot import name 'url_quote' from 'werkzeug.urls'`


## Details of the Fix
- Ensures `wh_receiver` can run on the latest compatible versions of dependencies, contributing to stability in the aconf core.
- Highlights the necessity for Flask and Werkzeug to maintain version compatibility, especially across major releases.

## Changes Made
- Freeze Werkzeug version to avoid compatibility issues with Flask 2.1.1. The application is verified stable with Werkzeug versions 2.2.2 and 2.3.7.
- Issues are expected with Werkzeug versions +3.x.x; recommend caution with future updates.

## Context for Changes
- The changes are crucial for maintaining a stable aconf core, given the ongoing updates to Android 9 on ATVs, leading to the recommendation not to update Flask at the current stage.
- Refer to conversations highlighted in [Pogo Tools Discord](https://discord.com/channels/639191866822098944/681977009055989940/1166837948268941412).

## Future Considerations
- Suggests a detailed review of these dependencies when the project priorities turn towards dependency updates. Significant enhancements or fixes in major releases could be beneficial for aconf.
- Noteworthy that Werkzeug's release history on [PyPI Werkzeug history](https://pypi.org/project/Werkzeug/#history) marks 2.3.7 as the final 2.x.x release on 14 August 2023 before transitioning to 3.x.x versions, which could introduce breaking changes.

## Testing
- Rapid local testing was conducted in Docker, confirming that aconf experiences no runtime errors and functions as intended with Werkzeug versions 2.2.2 and 2.3.7.
- Performance observed by running `@G3neRaL` on bare metal.

## Request for Feedback/Review
- Insights or reviews are highly welcomed.
- Discussions are open regarding potential risks or additional considerations that may not have been previously identified.

matt-yml committed:
freeze Werkzeug version at 2.3.7
* this is the latest version that aconf can be run with
* fixes Flask 2.1.1 runtime error:
`ImportError: cannot import name 'url_quote' from 'werkzeug.urls`

Flask and Werkzeug need to be kept in sync over major versions

matt-yml committed:
Merge branch 'TechG3n:rom_A9' into rom_A9

TechG3n commented:
This worked for in non docker install, can you test these?

```
configparser==5.2.0
Flask==2.2.5
PyMySQL==1.0.2
requests==2.31.0
mysql==0.0.3
mysql-connector-python==8.0.31
mysqlclient==2.1.0
Werkzeug==2.3.7
```

matt-yml committed:
update additional packages per @techG3n github dependency bot
recommendations for security issues.

https://github.com/TechG3n/aconf/pull/59#issuecomment-1783787655

matt-yml committed:
Merge branch 'rom_A9' of github.com:matt-yml/aconf-dev into rom_A9

matt-yml commented:
Thanks for running the bare metal test. I'm 100% containerized and don't have a spare bm for that. 
Was thinking of firing up an EC2 to test that, but that's still a VM.

Used @TechG3n 's proposed "requirements.txt". 
Build looks good, running fine after several minutes. 

Definitely clears the original error from the PR's initial problem.

Here're all the packages running in a known good state:
```
[deve@len01 volumes]$ dcp exec -it aconf bash
root@5c1d640dbdf2:/usr/src/app# pip freeze
certifi==2023.7.22
charset-normalizer==3.3.1
click==8.1.7
configparser==5.2.0
Flask==2.2.5
idna==3.4
itsdangerous==2.1.2
Jinja2==3.1.2
MarkupSafe==2.1.3
mysql==0.0.3
mysql-connector-python==8.0.31
mysqlclient==2.1.0
protobuf==3.20.1
PyMySQL==1.0.2
requests==2.31.0
urllib3==2.0.7
Werkzeug==2.3.7
root@5c1d640dbdf2:/usr/src/app#
```
